### ISSUE TITLE
skipping some transactions in UndoManager

### ISSUE THREAD

**Issue by calibr** [y-js/yjs#157]
I think it can be useful to make UndoManager ignore some transactions, for example:

```
undoManager.startSkipping()
yText.applyDelta([{
  insert: 'this text can not be undone'
}])
undoManager.stopSkipping()
```

So any changes between `startSkipping` and `stopSkipping` won't be undone

@dmonad WDYT?

**Comment by dmonad** [comment]
I recommend using the transaction-origin to selectively capture operations because this will lead to a cleaner API. Better than doing `start(); ..; stop()` all the time - at least this is what I believe. But I don't want to impose my opinions on the API and this sounds like a feature that will be requested very often in the future. 

**Recommended method:**

```js
// This won't be captured by the UndoManager
doc.transact(() => {
  yText.applyDelta([{
    insert: 'this text can not be undone'
  }])
}, 'api') // use anything as the transaction origin
```

Does this apply to your use-case too? Or do you think that `start` `stop` might still be useful?

**Comment by calibr** [comment]
@dmonad yes it looks like it works for me, thank you, I did not know about using transaction origin for this

**Comment by dmonad** [comment]
Let me know what you think about this concept and if it works nicely for you. There are a few examples of origin-filtering in the docs. I found that the transaction origins really help when different components interact with each other. 

.

**Comment by calibr** [comment]
@dmonad ok, so far it does work fine. We are going to test this and the latest yjs update more and I will let you know if we have any problems with this. thank you again:)
